src/main/java/at/hannibal2/skyhanni/features/garden/pests/PesthunterProfit.kt

```diff
@@ -0,0 +1,200 @@
+package at.hannibal2.skyhanni.features.garden.pests
+
+import at.hannibal2.skyhanni.config.ConfigUpdaterMigrator
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.features.garden.GardenAPI
+import at.hannibal2.skyhanni.features.inventory.patternGroup
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.test.command.ErrorManager
+import at.hannibal2.skyhanni.utils.DisplayTableEntry
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemPriceUtils.getPrice
+import at.hannibal2.skyhanni.utils.ItemUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.getInternalName
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.ItemUtils.itemName
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.NumberUtil.shortFormat
+import at.hannibal2.skyhanni.utils.RegexUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+@SkyHanniModule
+object PesthunterProfit {
+
+    data class PesthunterTrade(val internalName: NEUInternalName, val coinsPerPest: Double)
+    var bestPesthunterTrade = mutableListOf<PesthunterTrade>()
+
+    private val config get() = GardenAPI.config.pests.pesthunterShop
+    private var display = emptyList<Renderable>()
+
+    var inInventory = false
+
+    /**
+     * REGEX-TEST: §2100 Pests
+     */
+    private val pestCostPattern by patternGroup.pattern(
+        "garden.pests.inventory.cost",
+        "§2(?<pests>\\d+) Pests"
+    )
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        inInventory = false
+    }
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!config.pesthunterProfitEnabled) return
+        if (event.inventoryName != "Pesthunter's Wares") return
+
+        inInventory = true
+
+        val table = mutableListOf<DisplayTableEntry>()
+        for ((slot, item) in event.inventoryItems) {
+            try {
+                readItem(slot, item, table)
+            } catch (e: Throwable) {
+                ErrorManager.logErrorWithData(
+                    e, "Error in PesthunterProfit while reading item '${item.itemName}'",
+                    "item" to item,
```

Comment:
there is no point putting an items stack into exta data unless you know what Skull@9a7bab28 means